worker/src/main/scala/com/lucidchart/piezo/ConnectionProvider.scala

```diff
@@ -1,25 +1,131 @@
 package com.lucidchart.piezo
 
-import org.quartz.utils.HikariCpPoolingConnectionProvider
+import java.sql.{Connection, SQLTransientConnectionException}
 import java.util.Properties
+import java.util.concurrent.TimeUnit
+import org.quartz.utils.HikariCpPoolingConnectionProvider
 import org.slf4j.LoggerFactory
 
-class ConnectionProvider(props: Properties) {
+class ConnectionProvider(props: Properties, causeFailoverEveryConnection: Boolean = false) {
```

Comment:
This is a variable used only for tests. Since dependency injection is not included in this project, this is set as a default to false but can be overridden in the `JobHistoryModel.scala` to ensure that this class can successfully create new configuration pools when IP addresses to a host change

Reply:
I agree with both of Thayne's thoughts here. We should put this in a property so we don't put test logic in the application